docs_src/design/adr/core/0019-EdgeX-CLI-V2.md

```diff
@@ -0,0 +1,113 @@
+# EdgeX-CLI V2 Design
+ 
+
+## Status
+
+**Draft**
+
+## Context
+This ADR presents a technical plan for creation of a 2.0 version of the edgex-cli which supports the new V2 REST APIs developed as part of the Ireland release of EdgeX.
+
+## Existing Behavior
+The latest version of edgex-cli  (1.0.1) only supports the V1 REST APIs and thus cannot be used with V2 releases of EdgeX.
+
+As the edgex-cli was developed organically over time, the current implementation has a number of bugs mostly involving a lack of consistent behavior, especially with respect to formatting of output.
+
+Other issues with the existing client include:
+
+- lack of tab completion
+- default output of commands is too verbose
+- verbose output sometime prevents use of jq 
+- static configuration file required (i.e. no registry support)
+- project hierarchy not conforming to best practice guidelines 
+
+## History
+The original Hanoi V1 client was created by a team at VMWare which is no longer participating in the project. 
+Canonical will lead the development of the Ireland/Jakarta V2 client.
+
+## Decision
+
+1. Use standardized command-line args/flags
+
+| Argument/Flag      | Description |
+| ----------- | ----------- |
+| `-d`, `--debug`      | show additional output for debugging purposes (e.g. REST URL, request JSON, …). This command-line arg will replace -v, --verbose and will no longer trigger output of the response JSON (see -j, --json).       |
+| `-j`, `--json`   | output the raw JSON response returned by the EdgeX REST API and *nothing* else. This output mode is used for script-based usage of the client.    |
+| `--version`   | output the version of the client and if available, the version of EdgeX installed on the system (using the version of the core data service)   |
+
+
+2. Restructure the Go code hierarchy to follow the [most recent recommended guidelines](https://github.com/golang-standards/project-layout). For instance /cmd should just contain the main application for the project, not an implementation for each command - that should be in /internal/cmd
+
+3. Take full advantage of the features of the underlying command-line library, [Cobra](https://github.com/spf13/cobra), such as tab-completion of commands.
+
+4. Change the client binary name from `edgex-cli` to `edgex`. The reason is that the "-cli" is superfluous. There is no `edgex` tool and best practice is to choose a simple, memorable word. CLI tool names do not generally include `"-cli"` in the name. (Examples: `go`,`snap`,`kubectl`,`git`,`docker` etc)
```

Comment:
I am in the camp of removing '-cli'.  Other frameworks/tools would not include "cli" unless there were multiple types of clients to interact with.  To @lenny-intel - while explicit, what else would it be confused with?

Reply:
Would folks run the command `edgex` and expect the services to start?

Reply:
@lenny-intel / @jpwhitemn - I am on the fence on this one. Whilst I think the "-cli" might be superfluous, I see the point about `edgex-cli` being explicit and removing it might give the mistaken impression that running `edgex` by itself would be a way to start up the core services. Also, as we have `edgex-ui` it makes sense to have a corresponding `edgex-cli`. I have therefore removed this suggestion and suggest we stick with `edgex-cli` and don't change it for now.

Reply:
Works for me :-)